.github/workflows/deploy-branch.yml

```
@@ -0,0 +1,46 @@
+name: Deploy-dev
+
+on:
+  pull_request:
```

Shouldn't work on each PR, but on manual dispatch instead.
```suggestion
  workflow_dispatch:
```